karapace/backup/api.py

```diff
@@ -298,7 +298,13 @@ def _consume_records(
     if start_offset >= end_offset:
         raise EmptyPartition
 
-    end_offset -= 1  # high watermark to actual end offset
+    # kafka-python returns end offset + 1, i.e. the value that will be assigned
+    # to the next record to be produced. To get the highest offset already
+    # produced we need to subtract 1. Note that this has little to do with high
+    # watermark, which takes into account the highest record offset to be fully
+    # in sync across ISRs, and therefore can be an arbitrary number less than
+    # or equal to the LEO.
```

Comment:
great, thanks for the explaination! This save a lot of time to the reader of the code. Can we mention also what happens if the topic its empty?

Reply:
That would be good, I'll leave that as a room for further future improvement though.